Solutions/Marain.Workflow.Abstractions/Marain.Workflow.Abstractions.csproj

```diff
@@ -22,8 +22,8 @@
     <PackageReference Include="Corvus.Identity.ManagedServiceIdentity.ClientAuthentication" Version="0.2.0-preview.5" />
     <PackageReference Include="Corvus.Leasing.Abstractions" Version="0.5.0-preview.5" />
     <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.2.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.2" />
```

Comment:
Be careful with this. Functions v3 is still on v2.x bits for lots of `Microsoft.Extensions.*` pieces, so I have been studiously avoiding upgrading any leaf projects to 3.1. (Specs and function hosts are fine, as long as it works; I don't know offhand whether you can actually upgrade these two components to V3.1 on Functions, since Functions uses older versions of these components for its own purposes.)

Reply:
So would 2.2.x be a better choice? I will shift them back on that basis.

Reply:
Yes, 2.2.x is better. (Applications using these libraries will remain free to upgrade the refs to 3.1 if they choose.)